RandBLAS/trig_skops.hh

```diff
@@ -316,125 +298,157 @@ namespace RandBLAS {
 
     template <typename T>
     void fht_dispatch(
-        bool left,
+        bool left, // Pre-multiplying?
         blas::Layout layout,
-        T* buff,
-        int64_t log_n,
         int64_t num_rows,
-        int64_t num_cols
+        int64_t num_cols,
+        int64_t log_n,
+        T* A
         )
     {
         if(left && layout == blas::Layout::ColMajor)
-            fht_left_col_major(buff, log_n, num_rows, num_cols);
+            fht_left_col_major(A, log_n, num_rows, num_cols);
         else if(left && layout == blas::Layout::RowMajor)
-            fht_left_row_major(buff, log_n, num_rows, num_cols);
+            fht_left_row_major(A, log_n, num_rows, num_cols);
         else if(!left && layout == blas::Layout::ColMajor)
-            fht_right_col_major(buff, log_n, num_rows, num_cols);
+            fht_right_col_major(A, log_n, num_rows, num_cols);
         else
-            fht_right_row_major(buff, log_n, num_rows, num_cols);
+            fht_right_row_major(A, log_n, num_rows, num_cols);
     }
 }
```

Comment:
Please accept num_rows, num_cols, and a leading dimension parameter "lda". (You'll probably need the lda to also appear in the signatures of the functions you're dispatching here.)